share/github-backup-utils/ghe-backup-config

```diff
@@ -66,6 +66,26 @@ for f in "$GHE_BACKUP_CONFIG" "$GHE_BACKUP_ROOT/backup.config" \
   fi
 done
 
+# Check that moreutils ts has been installed
+if ! echo "moreutils" | ts &>/dev/null; then
+  echo "Error: moreutils not found. Please install https://joeyh.name/code/moreutils" 1>&2
+  exit 1
+fi
+
+ghe_parallel_check() {
+  # Some machines may have both moreutils parallel and GNU parallel installed.
+  parallel_command="parallel"
+  if [ -x "/usr/bin/parallel.moreutils" ]; then
+    parallel_command="/usr/bin/parallel.moreutils"
+  fi
+
+  # Check that the parallel_command is pointing to moreutils parallel
+  if ! $parallel_command -h | grep -q "parallel \[OPTIONS\] command -- arguments"; then
+    echo "Error: moreutils not found. Please install https://joeyh.name/code/moreutils" 1>&2
+    exit 1
+  fi
+}
+
```

Comment:
I wasn't very clear here earlier either, sorry. What I had in mind was combining these two together, ala:

```suggestion
# Check that moreutils has been installed
ghe_parallel_check() {
  if ! echo "moreutils" | ts &>/dev/null; then
    echo "Error: moreutils not found. Please install https://joeyh.name/code/moreutils" 1>&2
    exit 1
  fi
  
  # Some machines may have both moreutils parallel and GNU parallel installed.
  parallel_command="parallel"
  if [ -x "/usr/bin/parallel.moreutils" ]; then
    parallel_command="/usr/bin/parallel.moreutils"
  fi

  # Check that the parallel_command is pointing to moreutils parallel
  if ! $parallel_command -h | grep -q "parallel \[OPTIONS\] command -- arguments"; then
    echo "Error: moreutils not found. Please install https://joeyh.name/code/moreutils" 1>&2
    exit 1
  fi
}
```

This doesn't have to be wrapped in a function and can be exec'd directly.

Reply:
I tried calling these checks directly and not wrapping them in a function originally, but several of the inner scripts were failing to find `parallel` in it's path and was prompting for the installation of `moreutils`. This was causing `ghe-backup-pages`, etc. to fail. 

Plus there's no reason for every `ghe-backup-*` script that's invoked from within `ghe-backup` to execute these checks. The same especially goes for all of the `ghe-restore` scripts since restore is not using parallel at all. Wrapping these checks in a function that can be called from `ghe-restore` once we decide to go that route seems to me like the best compromise. Would you agree?

Reply:
Ah yeah. Good point.